js/ui/html_editor/utils/templates_storage.js

```diff
@@ -0,0 +1,39 @@
+import { isDefined } from '../../../core/utils/type';
+
+export default class TemplatesStorage {
+    constructor() {
+        this._widgetsStorage = {};
+    }
+
+    set({ widgetKey, marker }, value) {
+        let widgetTemplates = this._widgetsStorage[widgetKey];
+        if(!widgetTemplates) {
+            this._widgetsStorage[widgetKey] = widgetTemplates = widgetTemplates = {};
+        }
+
+        widgetTemplates[marker] = value;
+    }
+
+    get({ widgetKey, marker }) {
+        if(isDefined(widgetKey)) {
```

Comment:
we can implement this function this way
```javascript
const isQuillFormatCall = !isDefined(editorKey);
return isQuillFormatCall
    ? Object.values(this._storage).at(-1)[marker] // NOTE: ...
    : this._storage[editorKey][marker];
```

Reply:
we need have check on this._storage[editorKey]

Reply:
we can write `this._storage[editorKey]?.[marker];